src/transformers/trainer.py

```diff
@@ -1193,7 +1194,13 @@ def train(
                 raise ValueError(f"No valid checkpoint found in output directory ({args.output_dir})")
 
         if resume_from_checkpoint is not None:
-            if not os.path.isfile(os.path.join(resume_from_checkpoint, WEIGHTS_NAME)):
+            # SMP partial checkpoints are in {filename}_{pp_rank()}_{tp_rank()} or {filename}_{pp_rank()}_{tp_rank()}_{rdp_rank()} format.
+            checkpoint_file_exists = (
+                glob.glob(os.path.join(resume_from_checkpoint, WEIGHTS_NAME) + "_*")
+                if is_sagemaker_mp_enabled()
+                else os.path.isfile(os.path.join(resume_from_checkpoint, WEIGHTS_NAME))
+            )
```

Comment:
This is used several times, could we refactor it in a util function that takes the filename?